hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/math/RandFunctionIntegrationTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.math;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.sql.SqlColumnType;
+import com.hazelcast.sql.SqlErrorCode;
+import com.hazelcast.sql.SqlException;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.impl.SqlTestSupport;
+import com.hazelcast.sql.support.expressions.ExpressionValue;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class RandFunctionIntegrationTest extends SqlTestSupport {
+
+    private final TestHazelcastInstanceFactory factory = new TestHazelcastInstanceFactory(1);
+    private HazelcastInstance member;
+    private IMap map;
+
+    @Before
+    public void before() {
+        member = factory.newHazelcastInstance();
+
+        map = member.getMap("map");
+    }
+
+    @After
+    public void after() {
+        factory.shutdownAll();
+    }
+
+    @Test
+    public void testNoArg() {
+        map.put(0, 0);
+
+        double res1 = execute("");
+        double res2 = execute("");
+
+        assertNotEquals(res1, res2);
```

Comment:
We miss a test that two invocations of `rand()` in a single query return different value. It can be in a single row or in two rows or in the where clause: to sample 10% of rows one would use `WHERE rand()<0.1`

Reply:
Added a test

Reply:
Shouldn't we also test `SELECT rand(), rand() FROM map`? Maybe also in the WHERE clause.

Reply:
I do not think we need it. The function doesn't depend on the context of a row or specific placement in the query.

Reply:
I know it will work now, but the purpose of the test is to document the behavior so that it doesn't change in the future due to some optimization. It's easy to write such a test.

Reply:
Extended the test to invoke the function several times for a row